src/main/java/com/breskul/bibernate/util/EntityUtil.java

```diff
@@ -47,26 +133,139 @@ public static Field findEntityIdField(List<Field> fields) {
+    validateIsEntity(cls);
+    return findEntityIdField(getClassColumnFields(cls));
+  }
+
+
+  /**
+   * Composes a SELECT block for the specified list of column names.
+   *
+   * @param columnNames - The list of column names
+   * @return The composed SELECT block
+   */
   public static String composeSelectBlockFromColumns(List<Field> columnNames) {
     return columnNames.stream()
         .map(EntityUtil::resolveColumnName)
         .collect(Collectors.joining(", "));
   }
 
-  // TODO: javadoc
-  public static String resolveColumnName(Field idField) {
-    if (idField.isAnnotationPresent(Column.class)) {
-      return idField.getAnnotation(Column.class).name();
+
+  /**
+   * Resolves the column name for the given field, from field annotations or return field name if
+   * column annotation is absent.
+   *
+   * @param field - The field for which to resolve the column name
+   * @return The resolved column name
+   */
+  public static String resolveColumnName(Field field) {
+    if (field.isAnnotationPresent(Column.class)) {
+      var columnName = field.getAnnotation(Column.class).name();
+      return columnName.isBlank() ? field.getName() : columnName;
+    } else if (field.isAnnotationPresent(JoinColumn.class)) {
+      var columnName = field.getAnnotation(JoinColumn.class).name();
+      return columnName.isBlank() ? field.getName() + "_id" : columnName;
     }
-    return idField.getName();
+    return field.getName();
   }
 
+  /**
+   * Retrieves the join column name for joined entity.
+   *
+   * @param entityType    - The entity type
+   * @param joinedEntity  - The joined entity
+   * @return The join column name
+   * @throws IllegalStateException if the related entity field cannot be found in the entity type
+   */
+  public static String getJoinColumnName(Class<?> entityType, Class<?> joinedEntity) {
+    var joinField = Arrays.stream(entityType.getDeclaredFields())
+      .filter(field -> field.getType().equals(joinedEntity))
+      .findFirst()
+      .orElseThrow(() -> new IllegalStateException("Can't find related entity [%s] field in [%s]."
+        .formatted(joinedEntity, entityType)));
+
+    return resolveColumnName(joinField);
+  }
+
+
+  /**
+   * Retrieves the ID value of the given entity.
+   *
+   * @param entity - The entity object
+   * @return The ID value of the entity object
+   * @throws EntityParseException if the entity does not define an ID column or if multiple fields
+   *         are marked with the 'Id' annotation
+   */
   public static Object getEntityId(Object entity) {
     var idField = findEntityIdField(List.of(entity.getClass().getDeclaredFields()));
     return readFieldValue(entity, idField);
   }
 
+
+  /**
+   * Checks if the given field represents a primitive column in an entity.
+   *
+   * @param field - The field to check
+   * @return true if the field is a primitive column, false otherwise
+   */
+  public static boolean isPrimitiveColumn(Field field) {
```

Comment:
`isPrimitiveColumn` name is ambiguous, it might refer to primitive values only. How about `isSimpleColumn` or something in that direction?